CORE-3394 Simplifiy Assessor User Experience

plamut opened the pull request:
This PR adds a new and distraction-free `/assessments_view` view that only displays thel Assessments assigned to the current user's, without any other unnecessary page elements such as LHN, the widgets bar etc.

The PR also lints the `apps/dashboard.js` file, thus I recommend reviewing this on a commit-to-commit basis to see what has changed in there in terms of business logic.

Smotko commented:
We are probably going to need handle the state when there are no assessments mapped a bit better. Currently it a bit broken broken:

![capture 2016-04-01 at 17 48 55](https://cloud.githubusercontent.com/assets/513444/14433314/1fc8068a-000e-11e6-8c6e-72b8b7b45b07.png)

@plamut can you talk to @vladan-m / Predrag to figure out how to handle this. Also we need to create a ticket or talk to vladan to fix some white space issues.

plamut commented:
@vladan-m Ping me tomorrow please or send me an email. I need to know what to display if there are no Assessments assigned to the current user. For the whitespace and other minor styling-related polishing you can either send a PR directly to my branch, or I can open a separate ticket for that.

(perhaps the second options is better, because the non-essential styling fixes will not block this PR...)

vladan-m commented:
Hey @plamut and @Smotko currently we have `zero-state` in our app. It is very simple notification that is telling user that something still don't exist. Structure how it should be implemented is like this:
``` html
<div class="zero-state">
    <h2>
      You have no assigned tasks</h2>
  </div>
```

This is the list of the files where you can find `zero-state`. Please note that it doesn't matter if you are gonna put class `zero-state` on `<li>` or `<div>` element. It will work, so add it depends on your structure:
`src/ggrc_workflows/assets/mustache/cycle_task_group_object_tasks/tree_footer.mustache`
`src/ggrc_workflows/assets/mustache/dashboard/info/my_audit_requests.mustache`
`src/ggrc_workflows/assets/mustache/dashboard/info/my_tasks.mustache`
`src/ggrc_workflows/assets/mustache/dashboard/info/workflow_progress.mustache`

Let me know if you have any additional questions.

P.S. Regarding white space issues. If you are referring to white space inconsistency on assessment info page that one is solved yesterday. [This is PR](https://github.com/google/ggrc-core/pull/3740)

plamut commented:
@vladan Thanks for the pointers, I have used `zero-state` class. You can now (or when it's merged) look at the remaining spacing issues (if any) and polish them.

@Smotko Done. I configured the `Person --> Assessment` to use the `assessments/tree_header.mustache` template instead of the default one from `base_objects`. The former template was not used anywhere, thus I replaced it with the content of the existing `base_objects/tree_header.mustache`, and added the check for an empty list of Assessments.

vladan-m commented:
@plamut Awesome. I'll check tomorrow to see if this is merged and polish spacings is necessary.

hyperNURb commented:
👍 Looks good overall.
Should we want for @vladan-m to make CSS fixes or we should merge and he'll do the fixes in separate PR?

Smotko commented:
If we make a ticket for @vladan-m we can merge this now.

plamut commented:
I'll do it. No sense in blocking this because of the trivial styling fixes IMO.

hyperNURb commented:
👍
